elasticsearch-indexing/index.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+set -eo pipefail
+
+# Help
+function help(){
+  printf "\n"
+  printf "Usage: export <options> $0\n"
+  printf "\n"
+  printf "Options supported, export them as environment variables:\n"
+  printf "\t ES_SERVER=str,                    str=elasticsearch server url, default: ""\n"
+  printf "\t ES_USER=str,                      str=elasticsearch user, default: ""\n"
+  printf "\t ES_INDEX=str,                     str=elasticsearch index, default: perf_scale_ci\n"
+  printf "\t JENKINS_USER=str,                 str=Jenkins user, default: ""\n"
+  printf "\t JENKINS_API_TOKEN=str,            str=Jenkins API token to authenticate, default: ""\n"
+  printf "\t JENKINS_BUILD_TAG=str,            str=jenkins job build tag, it's a built-in env var and is automatically set in Jenkins environment\n"
+  printf "\t JENKINS_NODE_NAME=str,            str=jenkins job build tag, it's a built-in env var and is automatically set in Jenkins environment\n"
+  printf "\t JENKINS_BUILD_URL=str,            str=jenkins job build url, it's a built-in env var and is automatically set in Jenkins environment\n"
+}
+
+# Defaults
+if [[ -z $ES_SERVER ]]; then
+  echo "Elastic server is not defined, please check"
+  help
+  exit 1
+fi
+if [[ -z $ES_USER ]]; then
+  export ES_USER=""
+else
+  export ES_USER="--user ${ES_USER}"
+fi
+if [[ -z $ES_INDEX ]]; then
+  export ES_INDEX=perf_scale_ci
+fi
+if [[ -z $JENKINS_USER ]] || [[ -z $JENKINS_API_TOKEN ]]; then
+  echo "Jenkins credentials are not defined, please check"
+  help
+  exit 1
+fi
+
+# Generate a uuid
+export UUID=$(uuidgen)
+
+# Elasticsearch and jenkins credentials
+export ES_SERVER=$ES_SERVER
+export ES_USER=$ES_USER
+export ES_INDEX=$ES_INDEX
+export JENKINS_USER=$JENKINS_USER
+export JENKINS_API_TOKEN=$JENKINS_API_TOKEN
+
+# Jenkins job info
+export BUILD_TAG=$JENKINS_BUILD_TAG
+export NODE_NAME=$JENKINS_NODE_NAME
+export BUILD_URL=$JENKINS_BUILD_URL
+
+# Timestamp
+timestamp=`date +"%Y-%m-%dT%T.%3N"`
+
+# Get OpenShift cluster details
+cluster_name=$(oc get infrastructure cluster -o jsonpath='{.status.infrastructureName}')
+platform=$(oc get infrastructure cluster -o jsonpath='{.status.platformStatus.type}')
+masters=$(oc get nodes -l node-role.kubernetes.io/master --no-headers=true | wc -l)
+workers=$(oc get nodes -l node-role.kubernetes.io/worker --no-headers=true | wc -l)
```

Comment:
Not really sure how often it would come up (or if its actually an issue), however technically the workers can also be on the master nodes. Not sure if we want to exclude masters from the worker count so we don't count non-dedicated workers.